ocs_ci/helpers/helpers.py

```diff
@@ -6001,3 +6002,118 @@ def find_cephfilesystemsubvolumegroup(storageclient_uid=None):
+    """
+    Create NetworkFenceClass CR and verify Ips are populated
+    in respective CsiAddonsNode objects
+
+    """
+
+    logger.info("Creating NetworkFenceClass")
+    network_fence_class_dict = templating.load_yaml(constants.NETWORK_FENCE_CLASS_CRD)
+    network_fence_class_obj = create_resource(**network_fence_class_dict)
+    if network_fence_class_obj.ocp.get(
+        resource_name=network_fence_class_obj.name, dont_raise=True
+    ):
+        logger.info(
+            f"NetworkFenceClass {network_fence_class_obj.name} created successfully"
+        )
+
+    logger.info("Verifying CsiAddonsNode object for CSI RBD daemonset")
+    all_nodes = get_worker_nodes()
+
+    for node_name in all_nodes:
+        cidrs = get_rbd_daemonset_csi_addons_node_object(node_name)["status"][
+            "networkFenceClientStatus"
+        ][0]["ClientDetails"][0]["cidrs"]
+        assert len(cidrs) == 1, "No cidrs are populated to CSI Addons node object"
+        logger.info(f"Cidr: {cidrs[0]} populated in {node_name} CSI addons node object")
+
+
+def create_network_fence(node_name, cidr):
+    """
+    Create NetworkFence for the node
+
+    Args:
+        node_name (str): Name of the node
+        cidr (str): cidr
+
+    Returns:
+        OCS: NetworkFence object
+
+    """
+    network_fence_obj = OCP(
+        kind=constants.NETWORK_FENCE,
+        namespace=config.ENV_DATA["cluster_namespace"],
+        resource_name=node_name,
+    )
+
+    if not network_fence_obj.get(resource_name=node_name, dont_raise=True):
+        logger.info("Creating NetworkFence")
+        network_fence_dict = templating.load_yaml(constants.NETWORK_FENCE_CRD)
+        network_fence_dict["metadata"]["name"] = node_name
+        network_fence_dict["spec"]["cidrs"][0] = cidr
+        network_fence_obj = create_resource(**network_fence_dict)
+        if network_fence_obj.ocp.get(
+            resource_name=network_fence_obj.name, dont_raise=True
+        ):
+            logger.info(
+                f"NetworkFence {network_fence_obj.name} for node {node_name} created successfully"
+            )
+    else:
+        logger.info(f"Network fence object for {node_name} already exists!")
+    return network_fence_obj
+
+
+def unfence_node(node_name, delete=False):
+    """
+    Un-fence node
+
+    Args:
+        node_name (str): Name of the node
+        delete (bool): If True, delete the network fence object
+
+    """
+
+    network_fence_obj = OCP(
+        kind=constants.NETWORK_FENCE, namespace=config.ENV_DATA["cluster_namespace"]
+    )
+
+    if network_fence_obj.get(resource_name=node_name, dont_raise=True):
+        network_fence_obj.patch(
+            resource_name=node_name,
+            params='{"spec":{"fenceState":"Unfenced"}}',
+            format_type="merge",
+        )
+        assert (
+            network_fence_obj.get(resource_name=node_name)["spec"]["fenceState"]
+            != "Fenced"
+        ), f"{node_name} doesnt seem to be unfenced"
```

Comment:
can we make this message more  clear .like expected to be unfenced but its still fenced